modules/browsiBidAdapter.js

```diff
@@ -0,0 +1,171 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {config} from '../src/config.js';
+import {VIDEO} from '../src/mediaTypes.js';
+import {logError, logInfo, isArray, isStr} from '../src/utils.js';
+
+const BIDDER_CODE = 'browsi';
+const DATA = 'brwvidtag';
+const ADAPTER = '__bad';
+const USP_TO_REPLACE = '__USP__';
+const GDPR_STR_TO_REPLACE = '__GDPR_STR__';
+const GDPR_TO_REPLACE = '__GDPR__';
+export const ENDPOINT = 'https://rtb.avantisvideo.com/api/v2/auction/getbid';
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: 329,
+  supportedMediaTypes: [VIDEO],
+  /**
+   * Determines whether or not the given bid request is valid.
+   * @param bid
+   * @returns {boolean}
+   */
+  isBidRequestValid: function (bid) {
+    if (!bid.params) {
+      return false;
+    }
+    const {pubId, tagId} = bid.params
+    const {mediaTypes} = bid;
+    return !!(validateBrowsiIds(pubId, tagId) && mediaTypes?.[VIDEO]);
+  },
+  /**
+   * Make a server request from the list of BidRequests
+   * @param validBidRequests
+   * @param bidderRequest
+   * @returns ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const requests = [];
+    const {refererInfo, bidderRequestId, gdprConsent, uspConsent} = bidderRequest;
+    validBidRequests.forEach(bidRequest => {
+      const {bidId, adUnitCode, auctionId, transactionId, schain, params, sizes} = bidRequest;
+      const video = getVideoMediaType(bidRequest);
+
+      const request = {
+        method: 'POST',
+        url: params.endpoint || ENDPOINT,
+        data: {
+          requestId: bidderRequestId,
+          bidId: bidId,
+          timeout: getTimeout(bidderRequest),
+          baData: getData(),
+          referer: refererInfo.page || refererInfo,
+          gdpr: gdprConsent,
+          ccpa: uspConsent,
+          sizes: sizes,
```

Comment:
In the case of hybrid AdUnits (banner + video), sizes will refer to the banner size. If you would like to have `sizes` reference the video size you should use `video.playerSize` here.

Reply:
thank you - fixed